tests/integration/test_check_downloaded_output.py

```diff
@@ -39,11 +39,20 @@ def test_check_downloaded_output(test_env, default_request, tmpdir):
         tar.extractall(file_name)
 
     dependencies_path = path.join("deps", "gomod", "pkg", "mod", "cache", "download")
-    names = [i["name"] for i in response.data["dependencies"]]
-    for dependency in names:
-        package_name = utils.escape_path_go(dependency)
-        dependency_path = path.join(file_name, dependencies_path, package_name)
-        assert path.exists(dependency_path)
+    for dependency in response.data["dependencies"]:
```

Comment:
I think you could just only do the checks for the dependencies with the type of `gomod` and skip the dependencies with the type of `go-package`. This is okay in my opinion since the `gomod` dependencies are the source code that contain the `go-package` dependencies.

This would mostly keep the previous code and make the test simpler.